### DISCUSSION TITLE
Homogenise inputs from callable workflows and event inputs from dispatchable workflows

### DISCUSSION THREAD

**Discussion by umarcor** [community/community#9092] [Actions] [answered]
[pyTooling/Actions: Callable vs dispatchable workflows](https://github.com/pyTooling/Actions#callable-vs-dispatchable-workflows)

> Reusable Workflows are defined through the `workflow_call` event kind.
> Similarly, any "regular" Workflow can be triggered through a `workflow_dispatch` event.
> Both event kinds support `input` options, which are usable within the Workflow.
> Therefore, one might intuitively try to write a workflow which is both callable and dispatchable.
> In other words, which can be either reused from another workflow, or triggered through the API.
> Unfortunately, that is not the case.
> Although `input` options can be duplicated for both events, GitHub's backend exposes them through different objects.
> In dispatchable Workflows, the object is `${{ github.event.inputs }}`, while callable workflows receive `${{ inputs }}`.

> As a result, in order to make a reusable workflow dispatchable, a wrapper workflow is required.
> See, for instance, [hdl/containers: .github/workflows/common.yml](https://github.com/hdl/containers/blob/main/.github/workflows/common.yml) and [hdl/containers: .github/workflows/dispatch.yml](https://github.com/hdl/containers/blob/main/.github/workflows/dispatch.yml).

It would be desirable to expose dispatch inputs as `${{ inputs }}` or, the other way round, to make the reusable workflow inputs available as `${{ github.event.inputs }}`. That would allow to write reusable workflows which are also callable through the API.

**Comment by ethomson** [comment]
Thanks for the feedback.  We wouldn't want to change the `github.event`, since there could already be inputs there, but I can imagine exposing the `inputs` context to cut down on code duplication.

**Comment by chrispat** [comment, marked as answer]
We are working on a fix for this issue.  We will expose inputs from `workflow_dispatch` events into the `inputs` context in addition to the `github.event.inputs` context.

**Reply by nschmeller** [reply]
Thanks for the info! An ops question that I have, not sure if this is the right place to ask -- will this thread be updated if and when this fix comes out? Cheers!

**Reply by ajaykn** [reply]
@nschmeller We have rolled out the changes and here is the blog post as well
https://github.blog/changelog/2022-06-10-github-actions-inputs-unified-across-manual-and-reusable-workflows/

Now we should be able to see dispatch input are available in both `github.event.inputs` and `inputs` context.
Where as now `inputs` context is recommended to use, because we respect boolean types.
cc @chrispat , @avocaddo

**Reply by nschmeller** [reply]
Amazing, thank y'all so much!!!

**Reply by MartinLesterSynamedia** [reply]
For anyone else confused by empty values on GH Enterprise on 3.4 or lower this feature is in 3.5 (according to the docs).
Till then you need to stick with `github.event.inputs`
